Tests/MapboxMapsTests/Ornaments/InfoButton/MapboxInfoButtonOrnamentViewTests.swift

```diff
@@ -0,0 +1,115 @@
+        XCTAssertEqual(dontParticipateTitle, telemetryAlert.actions[1].title, "The second action should be a 'Don't Participate' button.")
+        telemetryAlert.tapButton(atIndex: 1)
+        XCTAssertFalse(infoButton.isMetricsEnabled, "Metrics should not be enabled after selecting 'Don't Participate'.")
+    }
+
+    func testTelemetryOptIn() throws {
+        UserDefaults.standard.set(false, forKey: MapboxInfoButtonOrnament.Constants.metricsEnabledKey)
+        let infoButton = MapboxInfoButtonOrnament()
+        let parentViewController = MockParentViewController()
+        parentViewController.view.addSubview(infoButton)
+        infoButton.infoTapped()
+
+        var infoAlert = try XCTUnwrap(parentViewController.currentAlert, "The info alert controller could not be found.")
+        XCTAssertNotNil(infoAlert)
+        infoAlert.tapButton(atIndex: 0)
+
+        var telemetryAlert = try XCTUnwrap(parentViewController.currentAlert, "The telemetry alert controller could not be found.")
+
+        XCTAssertEqual(telemetryAlert.actions.count, 3, "The telemetry alert should have 3 actions.")
+
+        let participatingTitle = NSLocalizedString("Participate", comment: "Telemetry prompt button")
+        XCTAssertEqual(participatingTitle, telemetryAlert.actions[2].title, "The third action should be a 'Participate' button.")
+
+        XCTAssertFalse(infoButton.isMetricsEnabled)
+
+        let dontParticipateTitle = NSLocalizedString("Don’t Participate", comment: "Telemetry prompt button")
+        XCTAssertEqual(dontParticipateTitle, telemetryAlert.actions[1].title, "The second action should be a 'Don't Participate' button.")
+
+        telemetryAlert.tapButton(atIndex: 2)
+        XCTAssertTrue(infoButton.isMetricsEnabled, "Metrics should be enabled after selecting 'Participate'.")
+
+        infoButton.infoTapped()
+        infoAlert = try XCTUnwrap(parentViewController.currentAlert, "The info alert controller could not be found.")
+        infoAlert.tapButton(atIndex: 0)
+        telemetryAlert = try XCTUnwrap(parentViewController.currentAlert, "The telemetry alert controller could not be found.")
+        let keepParticipatingTitle = NSLocalizedString("Keep Participating", comment: "Telemetry prompt button")
+        XCTAssertEqual(keepParticipatingTitle, telemetryAlert.actions[2].title, "The third action should be a 'Keep Participating' button.")
+        telemetryAlert.tapButton(atIndex: 2)
+        XCTAssertTrue(infoButton.isMetricsEnabled, "Metrics should be enabled after selecting 'Keep Participating'.")
+    }
+}
+
+class MockParentViewController: UIViewController {
+    var currentAlert: UIAlertController?
+
+    override func present(_ viewControllerToPresent: UIViewController, animated flag: Bool, completion: (() -> Void)? = nil) {
+        if let vc = viewControllerToPresent as? UIAlertController {
+            currentAlert = vc
+        }
+    }
+}
```

Comment:
Could we see if there's a way to avoid the need for this mock? Could we use UIViewController's `presentedViewController` property?

Reply:
Thanks! I did try that and the property returns `nil`, at least in the tests.

Reply:
I wonder if that's because the view controller under test hasn't been added to a window.